en/lessons/ecto/changesets.md

```diff
@@ -0,0 +1,372 @@
+  |> validate_length(:name, min: 2)
+  |> validate_number(:age, greater_than: 0, less_than: 120)
+  |> validate_name_appropriate()
+end
+```
+
+Now we can give it a try:
+
+```elixir
+iex> Person.changeset(%Person{}, %{"name" => "Offensive Name"})
+#Ecto.Changeset<
+  action: nil,
+  changes: %{},
+  errors: [name: {"contains prohibited words", []}],
+  data: #Person<>,
+  valid?: false
+>
+```
+
+No surprise to us, our new validation rule works!
+
+## Adding changes programatically
+
+Sometimes you want to introduce changes to a changeset manually.
+We previously saw how to use `changes/2` to apply a map of changes to our changeset, now we'll look at a different approach using `put_change/3`.
+
+To demonstrate `put_change/3` we'll update our `changeset/2` function to allow `:name` to be blank and when it's blank set it to `"Anonymous"`.
+In order to accomplish this let us create a new `set_anonymous/1` function that accepts and returns a changeset as we did earlier:
+
+```elixir
+def set_anonymous(changeset) do
+  name = get_field(changeset, :name)
+
+  if is_nil(name) or name == "" do
+    put_change(changeset, :name, "Anonymous")
+  else
+    changeset
+  end
+end
+```
+
+Nothing to difficult, right?  If the `:name` is blank or an empty string we want to put a change on to the changeset making `:name` equal to `"Anonymous"`.
+
+With our new function in place, let's update see what `changeset/2` looks like now:
+
+```elixir
+def changeset(struct, params) do
+  struct
+  |> cast(params, [:age, :name])
+  |> validate_required([:age])
+  |> validate_number(:age, greater_than: 0, less_than: 120)
+  |> validate_name_appropriate()
+  |> set_anonymous()
+end
+```
+
+Now if we don't pass `:name` then it will be set to `"Anonymous"` as expected:
+
+```elixir
+iex> Person.changeset(%Person{}, %{"age" => 99})
+#Ecto.Changeset<
+  action: nil,
+  changes: %{age: 99, name: "Anonymous"},
+  errors: [],
+  data: #Person<>,
+  valid?: true
+>
+```
+
+## Use-case specific changesets
+
+A single `changeset/2` function capturing all of the rules of our example application makes sense but for something larger it almost certainly does not.
+Having changeset functions with specific responsibilities is not uncommon.
+
+In an application with account registration you may have `registration_changeset/2` and later use `update_changeset/3` to apply updates to an existing user.
+
+## Schemaless changesets
+
+An exciting feature of `Ecto.Changeset` is the ability to apply changesets to more than just our Ecto schemas.
+With schemaless changes we can provide the same level of data validation and transformation, without the schema.
+
+While they are beyond the scope of these lessons, everything we need to know can be found in the official [schemaless changesets](https://hexdocs.pm/ecto/Ecto.Changeset.html#module-schemaless-changesets) docs.
+
+## Conclusion
+
+This is just the tip of the iceberg when it comes to the punch that `Ecto.Changeset`'s packs.
```

Comment:
Just a note that `tip of the iceberg` and `packs a punch` are common expressions, but might be hard to translate in other languages, making translator's job more difficult 🤔